Logger/Name.cs

```diff
@@ -0,0 +1,26 @@
+namespace Logger;
+
+
+public record Name(string FirstName, string? MiddleName, string LastName)
```

Comment:
Wow, this is really cool and well explained - small note I think the Guidelines recommend using record class for clarity
```suggestion
public record class Name(string FirstName, string? MiddleName, string LastName)
```